### PR TITLE
Github code scan fixes

### PR DESCRIPTION
**Pull request by jelly**
Let's first test if a pixel test still loads.

### CODE REVIEW

File: pkg/playground/speed.js [hunk 1]
```diff
@@ -176,12 +176,11 @@ function download(ev) {
 
     /* Allow use of HTTP URLs */
     if (path.value.indexOf("http") === 0) {
-        const anchor = document.createElement("a");
-        anchor.href = path.value;
+        const url = new URL(path.value);
```
Comment (cockpituous) [inline pkg/playground/speed.js:179]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18974-20230619-215725-36053af6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/playground/speed.js [hunk 2]
```diff
@@ -176,12 +176,11 @@ function download(ev) {
...
         options.payload = "http-stream2";
-        options.address = anchor.hostname;
-        options.port = parseInt(anchor.port, 10);
-        options.path = anchor.pathname;
+        options.address = url.hostname;
+        options.port = parseInt(url.port, 10);
+        options.path = url.pathname;
```
Comment (cockpituous) [inline pkg/playground/speed.js:183]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18974-20230619-215725-36053af6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
This changes a lot of pixel tests. E.g. [this looks like a regression](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18974-20230619-215725-36053af6-fedora-38/pixeldiff.html#TestStorageBasic-testBasic-partition-rtl), and so does [this](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18974-20230619-215725-36053af6-fedora-38/pixeldiff.html#TestStorageStratis-testBasic-fsys-row-rtl)

**Comment by martinpitt** [comment]
Oh, was the "HACK" comment meant to provoke these failed pixel diffs, and you only wanted to check that they are displayed correctly? The other three commits look great!

**Comment by jelly** [comment]
With the `HACK` commit we did get valid pixel tests https://cockpit-logs.us-east-1.linodeobjects.com/pull-18974-20230619-215725-36053af6-fedora-38/pixeldiff.html#TestStorageBasic-testBasic-partition-rtl

File: pkg/playground/speed.js (hunk 1 above)
Comment (cockpituous) [inline pkg/playground/speed.js:179]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18974-20230620-065303-a4180749-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/playground/speed.js (hunk 2 above)
Comment (cockpituous) [inline pkg/playground/speed.js:183]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18974-20230620-065303-a4180749-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**